File: 28-FindtheIndexoftheFirstOccurrenceinaString/28-FindtheIndexoftheFirstOccurrenceinaString.java
// Last updated: 8/6/2026, 9:15:59 AM
1class Solution {
2    public List<List<Integer>> generate(int numRows) {
3        if (numRows == 0) return new ArrayList<>();
4        if (numRows == 1) {
5            List<List<Integer>> result = new ArrayList<>();
6            result.add(Arrays.asList(1));
7            return result;
8        }
9        
10        List<List<Integer>> prevRows = generate(numRows - 1);
11        List<Integer> newRow = new ArrayList<>();
12        
13        for (int i = 0; i < numRows; i++) {
14            newRow.add(1);
15        }
16        
17        for (int i = 1; i < numRows - 1; i++) {
18            newRow.set(i, prevRows.get(numRows - 2).get(i - 1) + prevRows.get(numRows - 2).get(i));
19        }
20        
21        prevRows.add(newRow);
22        return prevRows;
23    }
24}

File: 28-FindtheIndexoftheFirstOccurrenceinaString/28-FindtheIndexoftheFirstOccurrenceinaString.py
# Last updated: 8/6/2026, 9:12:37 AM
1class Solution:
2  def addBinary(self, a: str, b: str) -> str:
3    s = []
4    carry = 0
5    i = len(a) - 1
6    j = len(b) - 1
7
8    while i >= 0 or j >= 0 or carry:
9      if i >= 0:
10        carry += int(a[i])
11        i -= 1
12      if j >= 0:
13        carry += int(b[j])
14        j -= 1
15      s.append(str(carry % 2))
16      carry //= 2
17
18    return ''.join(reversed(s))